### PR TITLE
Test validate.checkSigs

### PR DESCRIPTION
**Pull request by bh2smith**
These tests should cover all possible outcomes but still only getting 87.5% on the branch coverage. Not sure whats up but would like to get to the bottom of this. Wanted to avoid putting too many todos around the code, but I really think we should pass **inputCount** as a **Boolean** especially considering it is only able to hand the case for two inputs anyway. 

This will involve/require that we also change the uses of this function in the plasma contract. Not a huge refactoring, and could be done easily.

### CODE REVIEW

**Comment by josojo** [comment]
Yeah, I really don't know whether it is best to work with many outputs or just two. 
Sure, if we use more than 2, transactions get longer and everything uses more gas. But on the other hand side, we could later merge much better different outputs with just one transaction instead of many dependent transactions.

Hm... my gut feeling is that working with 2 outputs would be better...

**Review by josojo — APPROVED** [review]

File: test/utilities.js [hunk 1]
```diff
@@ -116,6 +116,27 @@ const rlpEncodeTransaction = function(exitor, token, amount, inputCount, oindex)
   return RLP.encode(list)
 }
 
+function keccak256(...args) {
```
Comment (fleupold) [inline test/utilities.js]: I believe this is equivalent to what the already imported sha3 function is doing. Can we try using this one instead?

File: test/validate.js [hunk 2]
```diff
@@ -25,6 +28,89 @@ contract("Validate", () => {
       await assertRejects(validator.checkSigs(zeroHash32, zeroHash32, 0, zeroSig195))
     })
 
-    // TODO - Get Validator to pass
+    it("Basic Pass & Fail - Single Input", async () => {
```
Comment (fleupold) [inline test/validate.js]: I tend to prefer test cases testing one thing, e.g. here `it("accepts valid signature with one input")` and `it("rejects invalid signature with one input")`, etc.

**Review by fleupold — CHANGES_REQUESTED** [review]
Can we use the same sha3 implementation we use for the plasma and merkle contract tests as well?

Is it possible to run a coverage analysis to identify which branches are not covered?

File: test/utilities.js (hunk 1 above)
Comment (bh2smith) [reply test/utilities.js]: I tried using this one and they differ. This non-standard JS library thing is bullshit.

Comment (bh2smith) [reply test/utilities.js]:
```
toHex(sha3([txHash, rootHash])) = 0x54a8c...
keccak256(txHash, rootHash) = 0xbd6aa...
```

Comment (bh2smith) [reply test/utilities.js]: I will, thus, change all occurrences of *web3.sha3* to *toHex(sha3)* but will leave keccak for items that need to be hashed together unless you have a better suggestion.

**Comment by bh2smith** [comment]
@fleupold  - ready to be checked again!

**Comment by fleupold** [comment]
I have submitted a suggestion on how it would work without the `keccak256`. In general I'm not against convenience functions, but this one we obviously copied from some stackoverflow answer and in it should do exactly what the `ethereumjs-util` `sha3` does.

Feel free to revert my changes, but in that case please replace all use of `ethereumjs-util` `sha3` with the new `keccak256`

**Review by fleupold — APPROVED** [review]

**Pull request merged**